desktopexporter/internal/server/server.go

```diff
@@ -22,57 +22,102 @@ var assets embed.FS
 
 type Server struct {
 	server http.Server
-	store  *store.Store
+	Store  *store.Store
 }
 
-func (s *Server) clearDataHandler() func(http.ResponseWriter, *http.Request) {
-	return func(writer http.ResponseWriter, request *http.Request) {
-		s.store.ClearTraces(request.Context())
-		writer.WriteHeader(http.StatusOK)
+func NewServer(endpoint string) *Server {
+	s := Server{
+		server: http.Server{
+			Addr: endpoint,
+		},
+		Store: store.NewStore(context.Background()),
 	}
+
+	serveFromFS, err := strconv.ParseBool(os.Getenv("SERVE_FROM_FS"))
+	if err != nil {
+		serveFromFS = false
+	}
+
+	s.server.Handler = s.Handler(serveFromFS)
+	return &s
 }
 
-func (s *Server) sampleDataHandler() func(http.ResponseWriter, *http.Request) {
-	return func(writer http.ResponseWriter, request *http.Request) {
-		sample := telemetry.NewSampleTelemetry()
-		s.store.AddSpans(request.Context(), sample.Spans)
+func (s *Server) Start() error {
+	defer s.Store.Close()
 
-		//TODO: Add sample logs and metrics
-		writer.WriteHeader(http.StatusOK)
+	_, isCI := os.LookupEnv("CI")
+	if !isCI {
+		go func() {
+			// Wait a bit for the server to come up to avoid a 404 as a first experience
+			time.Sleep(250 * time.Millisecond)
+			endpoint := s.server.Addr
+			browser.OpenURL("http://" + endpoint + "/")
+		}()
 	}
+	return s.server.ListenAndServe()
 }
 
-func writeJSON(writer http.ResponseWriter, data any) {
-	jsonTraceData, err := json.Marshal(data)
+func (s *Server) Close() error {
+	return s.server.Close()
+}
+
+func (s *Server) Handler(serveFromFS bool) http.Handler {
```

Comment:
`serveFromFS` should probably also be an option passed into `NewServer`. 

Naively, I would expect the routes to be set up as part of `NewServer` not require a further call

Reply:
I'm not sure where else to handle this one, since serveFromFS is only used during development. I'll figure it out in the same PR as the others though.